Neos.Neos/Classes/Domain/Service/SiteImportService.php

```diff
@@ -0,0 +1,92 @@
+use Neos\ContentRepository\Core\SharedModel\Workspace\WorkspaceName;
+use Neos\ContentRepository\Export\Factory\ContentRepositorySetupProcessorFactory;
+use Neos\ContentRepository\Export\Factory\EventStoreImportProcessorFactory;
+use Neos\ContentRepository\Export\ProcessingContext;
+use Neos\ContentRepository\Export\ProcessorInterface;
+use Neos\ContentRepository\Export\Processors\AssetRepositoryImportProcessor;
+use Neos\ContentRepository\Export\Severity;
+use Neos\ContentRepositoryRegistry\ContentRepositoryRegistry;
+use Neos\ContentRepositoryRegistry\Processors\ProjectionCatchupProcessor;
+use Neos\ContentRepositoryRegistry\Processors\ProjectionCatchupProcessorFactory;
+use Neos\ContentRepositoryRegistry\Service\ProjectionServiceFactory;
+use Neos\Flow\Annotations as Flow;
+use Neos\Flow\Persistence\Doctrine\Service as DoctrineService;
+use Neos\Flow\Persistence\PersistenceManagerInterface;
+use Neos\Flow\ResourceManagement\ResourceManager;
+use Neos\Flow\ResourceManagement\ResourceRepository;
+use Neos\Media\Domain\Repository\AssetRepository;
+use Neos\Neos\Domain\Import\DoctrineMigrateProcessor;
+use Neos\Neos\Domain\Import\LiveWorkspaceCreationProcessor;
+use Neos\Neos\Domain\Import\LiveWorkspaceIsEmptyProcessorFactory;
+use Neos\Neos\Domain\Import\SiteCreationProcessor;
+use Neos\Neos\Domain\Repository\DomainRepository;
+use Neos\Neos\Domain\Repository\SiteRepository;
+
+#[Flow\Scope('singleton')]
+final readonly class SiteImportService
+{
+    public function __construct(
+        private ContentRepositoryRegistry $contentRepositoryRegistry,
+        private DoctrineService $doctrineService,
+        private SiteRepository $siteRepository,
+        private DomainRepository $domainRepository,
+        private AssetRepository $assetRepository,
+        private ResourceRepository $resourceRepository,
+        private ResourceManager $resourceManager,
+        private PersistenceManagerInterface $persistenceManager,
+        private WorkspaceService $workspaceService,
+        private ProjectionCatchupProcessorFactory $projectionCatchupProcessorFactory,
+    ) {
+    }
+
+    /**
+     * @param \Closure(string): void $onProcessor Callback that is invoked for each {@see ProcessorInterface} that is processed
+     * @param \Closure(Severity, string): void $onMessage Callback that is invoked whenever a {@see ProcessorInterface} dispatches a message
+     */
+    public function importFromPath(ContentRepositoryId $contentRepositoryId, string $path, \Closure $onProcessor, \Closure $onMessage): void
+    {
+        if (!is_dir($path)) {
+            throw new \InvalidArgumentException(sprintf('Path "%s" is not a directory', $path), 1729593802);
+        }
+        $filesystem = new Filesystem(new LocalFilesystemAdapter($path));
+        $context = new ProcessingContext($filesystem, $onMessage);
+        $contentRepository = $this->contentRepositoryRegistry->get($contentRepositoryId);
+
+        // TODO make configurable (?)
```

Comment:
rather idea than todo ^^ :D

Reply:
Agree this can be removed. I also think that we do not need extensibility here as anyone can create its own processing chain. If he is able to create a custom processor he can aswell do that.